### DISCUSSION TITLE
Lucene syntax in HUNT

### DISCUSSION THREAD

**Discussion by peko19** [Security-Onion-Solutions/securityonion#6664] [Unsupported Versions] [answered]
hi,

How is it possible to construct a HUNT query so that in  ... | groupby  **STATEMENT** 

**STATEMENT** will be field with space/s in name - like for example in:

winlog.event_data.product Name

thanks

**Comment by dougburks** [comment]
Have you tried enclosing in quotes?
```
"winlog.event_data.product Name"
```

Or perhaps escaping the space with `\`?
```
winlog.event_data.product\ Name
```

**Comment by peko19** [comment]
yes, I tried these and some other too.
but nothing works like intended.

**Reply by dougburks** [reply]
> yes, I tried these and some other too.

It would be helpful if you include what you've tried so far.

Have you tried both quotes and escaping?
```
"winlog.event_data.product\ Name"
```

**Comment by peko19** [comment]
yes i tried "winlog.event_data.product\ Name"

and  so far all the hints from lucene syntax guides.

the best solution would be functional "groupby" from context menu  on  these fields with spaces.
wilnlogs are full of these.

**Comment by jertel** [comment, marked as answer]
I've confirmed this is a bug in SOC. There are two bugs actually:
1. Aggregated fields are being sent to ES with the quotes intact, and ES is not stripping them out.  This will be fixed in the next release.
2. The Hunt UI is not correctly detecting the quoted groupby fields as a single field and instead shows two chips, one for the term on the left side of the space, and another for the term on the right side. I'll investigate if this can be improved for the next release. But 1. will get you going either way.
